pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -894,6 +908,9 @@ func autoAllocateIPs(services []*model.Service) []*model.Service {
 		// 3. the hostname is not a wildcard
 		if svc.DefaultAddress == constants.UnspecifiedIP && !svc.Hostname.IsWildCarded() &&
 			svc.Resolution != model.Passthrough {
+			if svc.AutoAllocatedIPv4Address != "" {
```

Comment:
Yes, different ips can be allocated to a `hostname` in hash collision case. Previously we also face the same situation during service creation/deletion because different istiod can not handle same services state all at the same time.

> Would it cause split brain situation?

It would not influence service routing correctly even proxy sees different ips in bad situation.

Reply:
Proxy will change Istiod instances every 30min, if the IP allocation difers from different istiod then they will have issues every 30min

Reply:
I understand that dns and xds cannot take effect at the same time.  This is not ideal too